app/controllers/registration_commands_controller.rb

```diff
@@ -52,7 +52,11 @@ def host_groups_json
   end
 
   def smart_proxies
-    SmartProxy.with_features('Templates') & SmartProxy.with_features('Registration')
+    proxies = SmartProxy.with_features('Templates') & SmartProxy.with_features('Registration')
+    proxies.map do |proxy|
+      url = proxy.setting('Registration', 'registration_url').presence || proxy.url
```

Comment:
Some design debate: this already includes the registration commands concern. Can't we introduce a helper method `registration_url(proxy)` in the concern that does this and then reuse it in two places?

I also wonder if this URL should be the full endpoint or if it should be named `base_url`. Reading the JS it's only used in the helper text. I'd think it makes sense to provide the complete URL (including `/register`) to keep things consistent.